apps/web/pages/event-types/index.tsx

```diff
@@ -63,41 +63,83 @@ type EventTypeGroup = inferQueryOutput<"viewer.eventTypes">["eventTypeGroups"][n
 type EventType = EventTypeGroup["eventTypes"][number];
 interface EventTypeListProps {
   group: EventTypeGroup;
+  groupIndex: number;
   readOnly: boolean;
   types: EventType[];
 }
 
-export const EventTypeList = ({ group, readOnly, types }: EventTypeListProps): JSX.Element => {
+const Item = ({ type, group, readOnly }: any) => {
+  const { t } = useLocale();
+
+  return (
+    <Link href={"/event-types/" + type.id}>
+      <a
+        className="flex-grow truncate text-sm"
+        title={`${type.title} ${type.description ? `– ${type.description}` : ""}`}>
+        <div>
+          <span
+            className="truncate font-medium text-neutral-900 ltr:mr-1 rtl:ml-1"
+            data-testid={"event-type-title-" + type.id}>
+            {type.title}
+          </span>
+          <small
+            className="hidden text-neutral-500 sm:inline"
+            data-testid={"event-type-slug-" + type.id}>{`/${group.profile.slug}/${type.slug}`}</small>
+          {type.hidden && (
+            <span className="rtl:mr-2inline items-center rounded-sm bg-yellow-100 px-1.5 py-0.5 text-xs font-medium text-yellow-800 ltr:ml-2">
+              {t("hidden")}
+            </span>
+          )}
+          {readOnly && (
+            <span className="rtl:mr-2inline items-center rounded-sm bg-gray-100 px-1.5 py-0.5 text-xs font-medium text-gray-800 ltr:ml-2">
+              {t("readonly")}
+            </span>
+          )}
+        </div>
+        <EventTypeDescription eventType={type} />
+      </a>
+    </Link>
+  );
+};
+
+const MemoizedItem = React.memo(Item);
+
+export const EventTypeList = ({ group, groupIndex, readOnly, types }: EventTypeListProps): JSX.Element => {
   const { t } = useLocale();
   const router = useRouter();
 
   const utils = trpc.useContext();
   const mutation = trpc.useMutation("viewer.eventTypeOrder", {
-    onError: (err) => {
+    onError: async (err) => {
       console.error(err.message);
-    },
-    async onSettled() {
```

Comment:
Doing a refetch isn't really necessary in case of success. We manually update the cache below instead.